environs/config/config.go

```diff
@@ -1095,6 +1118,22 @@ func (c *Config) FanConfig() (network.FanConfig, error) {
 	return network.ParseFanConfig(c.asString(FanConfig))
 }
 
+// CloudInitUserData returns a copy of the raw user data attributes
+// that were specified by the user.
+func (c *Config) CloudInitUserData() (map[string]interface{}, error) {
```

Comment:
We generally don't return errors here. The validation is done earlier in the Validate() method.
See UpdateStatusHookInterval() for example.
We also tend to, for strings, return a bool to indicate whether the string has been set

Reply:
updated.